### PR TITLE
More user-friendly docs (Addressing Issue #1175)

### LINKED ISSUE #1175: Restructure documentation for user-friendliness
The [MDAnalysis documentation](http://pythonhosted.org/MDAnalysis/) is structured in such a way that it contains scant information on the [user interface](http://pythonhosted.org/MDAnalysis/documentation_pages/overview.html) and otherwise gives detail on the internal components of MDAnalysis that users should never really have to touch directly ([parsers](http://pythonhosted.org/MDAnalysis/documentation_pages/topology_modules.html), [readers and writers](http://pythonhosted.org/MDAnalysis/documentation_pages/coordinates_modules.html)). We've seen usage on the mailing list from users in which they try e.g. to directly use a `Reader` object to parse a trajectory, and the docs don't make it clear that this isn't really the way to go about things.

I think the docs should be restructured to have three parts. The first part should read more like the [MDAnalysis Tutorial](http://www.mdanalysis.org/MDAnalysisTutorial/), giving a friendly view as to *how* one uses `MDAnalysis` in practice. The second part should be an API reference, with the full sphinx autodoc output for the user interface components showed in part one. The last part should be the developer docs, which would include the reference docs for internal components, such as parsers, readers, and writers so that they are *somewhere*.

Thoughts?

### PR DESCRIPTION
**Pull request by richardjgowers**
Docs split into User guide, then API reference

Fixes #1175

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
So Issue #1175 seemed to get warmly received, so I've started an implementation of it.

I tried to come up with topics that covers everything to a moderate level, if anyone can think of more topics or a better way to order topics, then this is the PR for that!

**Comment by orbeckst** [comment]
That's a good structure. Now we just need some content... Should we essentially merge this as an incomplete PR and then raise Issues/PRs for things that are still incomplete? Or make PRs against this branch? In any case, I think it would be good to have sub-issues to encourage splitting the load.

**Comment by jbarnoud** [comment]
I would advocate against merging the PR as it, even to be filled latter on. This would break the develop branch, which we try to avoid as much as possible.

**Comment by richardjgowers** [comment]
We can PR onto this branch, then merge this when everything is done I guess

File: package/doc/sphinx/source/documentation_pages/bonds_angles_torsions.rst [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+     >>> u.atoms.guess_bonds(vdwradii={'C': 1.61, 'N':1.64}
+
+For large systems this will become very slow as it searches all pairwise
+combinations.  To circumvent this you can instead only search within a
+smaller group of atoms, for example looping through the segments in a
+system.
+::
+
+     for segment in u.segments:
+	 segment.atoms.guess_bonds()
```
Comment (orbeckst) [inline package/doc/sphinx/source/documentation_pages/bonds_angles_torsions.rst]: identation seems wrong

File: package/doc/sphinx/source/documentation_pages/advanced_reader.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.. -*- coding: utf-8 -*-
+
+.. All the advanced options in Reader usage and creation
```
Comment (orbeckst) [inline package/doc/sphinx/source/documentation_pages/advanced_reader.rst]: Also MemoryReader

File: package/doc/sphinx/source/documentation_pages/bonds_angles_torsions.rst (hunk 1 above)
Comment (kain88-de) [inline package/doc/sphinx/source/documentation_pages/bonds_angles_torsions.rst]: no it's correct on that line. 4 spaces it aligns as it should.

File: package/doc/sphinx/source/documentation_pages/coordinate_manipulation.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+     >>> from MDAnalysisTests.datafiles import PSF, DCD
+     >>> ag = u.select_atoms('name CA')
+     >>> ag.positions
+     array([[ 11.66462231,   8.39347267,  -8.98323059],
+            [ 11.41483879,   5.43442154,  -6.51348448],
+          ...
+     >>> vel = ag.velocities
+     NoDataError: Timestep does not contain velocities
+
+These properties return a **copy** of the data for the AtomGroup at the
```
Comment (kain88-de) [inline package/doc/sphinx/source/documentation_pages/coordinate_manipulation.rst]: Do really return a copy of the arrays? That seems wasteful and slow in terms of system memory. I thought it was a reference that is returned.

**Comment by jbarnoud** [comment]
It cannot be a reference: it is a non contiguous slice of the overall array.

**Comment by richardjgowers** [comment]
Yeah a numpy fancy index is always a copy, afaik we can't return a
reference/view

File: package/doc/sphinx/source/documentation_pages/coordinate_manipulation.rst (hunk 3 above)
Comment (orbeckst) [inline package/doc/sphinx/source/documentation_pages/coordinate_manipulation.rst]: I think we use fancy slicing on `ts._pos` so yes, it will be a copy.

**Comment by orbeckst** [comment]
@richardjgowers is this PR something we want to get into 0.16.x or should we re-target to 0.17.0?

**Comment by richardjgowers** [comment]
@orbeckst ideally it gets merged as soon as its finished.  This isn't code breaking/making so I don't think we're bound by version numbers.

File: package/doc/sphinx/source/index.rst [hunk 4]
```diff
@@ -67,16 +67,33 @@ literature. Please make sure to also reference any
...
+   ./documentation_pages/working_with_trajectories
+   ./documentation_pages/coordinate_manipulation
+   ./documentation_pages/bonds_angles_torsions
+   ./documentation_pages/advanced_universe
+   ./documentation_pages/advanced_reader
+   ./documentation_pages/advanced_topology
+
+
+API Reference
+=============
```
Comment (orbeckst) [inline package/doc/sphinx/source/index.rst]: With the sidebar navigation in #1126 I am pretty sure that we need a single toctree. I would suggest to keep the top-toctree pretty light, essentially

1. Overview
2. User guide 
3. API Reference
4. References

where _User guide_ is what you called "MDAnalysis Basics"

**Comment by orbeckst** [comment]
I am currently rebasing to develop and cleaning up the history. I will force-push to this branch.

**Comment by orbeckst** [comment]
I moved it to 0.17.0 – I don't think that anyone has currently got the bandwidth to make this happen for 0.16.2.

**Comment by orbeckst** [comment]
I moved it to 1.0. 

Obviously, any improvements to the docs are more than welcome at any time, but this should not hold up a 0.17.

**Comment by orbeckst** [comment]
Will the completion of the UserGuide by @lilyminium make this PR obsolete?

I think we still want to clean up the user docs and make them more API centric and have the UserGuide as the explanatory document – opinions?

**Comment by lilyminium** [comment]
@orbeckst As part of GSOD I was hoping to clean up the API docs as well, yes. I'm keeping an ongoing list of more significant doc issues in #2401

**Comment by richardjgowers** [comment]
I think this is obsolete thanks to @lilyminium

**Pull request closed without merging**